packages/core/test/perf/runners.js

```diff
@@ -65,6 +66,46 @@ function runSuite(suite) {
     .run();
 }
 
+function mostFromArray (a) {
+  return new FromArray(a)
+}
+
+const PropagateTask = require('../../src/scheduler/PropagateTask').PropagateTask
+const asap = require('../../../scheduler').asap
+
+class FromArray {
+  constructor (array) {
+    this.array = array
+  }
+  run (sink, scheduler) {
+    return asap(new FromArrayTask(this.array, sink), scheduler)
+  }
+}
+
+class FromArrayTask {
+  constructor (array, sink) {
+    this.array = array
+    this.sink = sink
+    this.active = true
+  }
+
+  run (t) {
+    for (var i = 0, l = this.array.length; i < l && this.active; ++i) {
```

Comment:
If I get this right, `this.active` is `true` until `dispose` is called. Is it possible that `this.sink.event` can somehow trigger `dispose` of the instance to be called? If not, would it make sense to check `this.active` _before_ the for-loop to return early? Then `this.active` could also be removed on L97.

Reply:
> Is it possible that this.sink.event can somehow trigger dispose of the instance to be called?

It is: take() and takeWhile() will dispose a stream immediately when they detect their termination conditions.  That will happen from inside the call stack that starts at `this.sink.event`.  So, to be absolutely correct, and to allow take to be used in perf tests, the loop has to check `this.active` on every turn of the loop, and before calling `this.sink.end`.

Did that make sense?

Reply:
Yes, it does. All good.